app/src/main/cpp/skyline/services/timesrv/time_shared_memory.cpp

```diff
@@ -0,0 +1,144 @@
+// SPDX-License-Identifier: MPL-2.0
+// Copyright © 2021 Skyline Team and Contributors (https://github.com/skyline-emu/)
+
+#include "core.h"
+#include "time_manager_server.h"
+
+namespace skyline::service::timesrv::core {
+    struct __attribute__((packed)) TimeSharedMemoryLayout {
+        template<typename T>
+        struct ClockContextEntry {
+            u32 updateCount;
+            u32 _pad_;
+            std::array<T, 2> context;
+        };
+
+        ClockContextEntry<SteadyClockTimePoint> standardSteadyClockContextEntry;
+        ClockContextEntry<SystemClockContext> localSystemClockContextEntry;
+        ClockContextEntry<SystemClockContext> networkSystemClockContextEntry;
+        struct __attribute__((packed)) {
+            u32 updateCount;
+            std::array<u8, 2> enabled;
+        } standardUserSystemClockAutomaticCorrectionEnabledEntry;
+    };
+    static_assert(offsetof(TimeSharedMemoryLayout, localSystemClockContextEntry) == 0x38);
+    static_assert(offsetof(TimeSharedMemoryLayout, networkSystemClockContextEntry) == 0x80);
+    static_assert(offsetof(TimeSharedMemoryLayout, standardUserSystemClockAutomaticCorrectionEnabledEntry) == 0xC8);
+
+    /**
+     * @brief Time Shared Memory uses a double buffered format that alternates writes context data, this is a helper to simplify that
```

Comment:
There should a clear explanation regarding why this function requires memory stores to be strongly ordered if not in the function comment in the function itself (`// The item value must be updated prior to updateCount to prevent reading in an invalid item value`), also fix `that alternates writes context data`.